psychopy/sound/backend_ptb.py

```diff
@@ -360,27 +362,6 @@ def statusDetailed(self):
             return None
         return self.track.status
 
-    @property
-    def status(self):
-        """status gives a simple value from psychopy.constants to indicate
-        NOT_STARTED, STARTED, FINISHED, PAUSED
-
-        Psychtoolbox sounds also have a statusDetailed property with further info"""
-
-        if self.__dict__['status']==STARTED:
-            # check portaudio to see if still playing
-            pa_status = self.statusDetailed
-            if not pa_status['Active'] and pa_status['State']==0:
-                # we were playing and now not so presumably FINISHED
-                self._EOS()
```

Comment:
One small caveat is that if `self.loops > 0`, then there will be a period of time if somehow `not pa_status['Active'] and pa_status['State'] == 0` evaluates to `True`, but `._loopsFinished` hasn't reached the target `.loops` number, then the above proposal will accumulate `._loopsFinished` by 1 every time `isFinished()` is called. But I guess this may be a ok behavior, since if a sound should be looping while `not pa_status['Active'] and pa_status['State'] == 0` is `True`, it likely indicates that a loop has finished. The existing solution with `@status.setter` isn't great either since it is gated by `self.__dict__['status']==STARTED:`.

Reply:
If nothing in _EOS is gated by `._isFinished`, what if we set `self._isFinished` before calling `_EOS()` rather than as an alternative to calling `_EOS()`? That way subsequent calls wouldn't duplicate.

So like this:
```
        # was the sound already finished?
        wasFinished = self._isFinished
        # is it finished now? (also sets _isFinished)
        isFinished = self._isFinished = not pa_status['Active'] and pa_status['State'] == 0
        # if it wasn't finished but now is, do end of stream behaviour
        if isFinished and not wasFinished:
            self._EOS()
```

Reply:
I guess one thing that is still unclear to me is how should `._isFinished` be set when there are **loops** specified? From reading `_EOS()`, I get the sense that `_isFinished` should only get turned to true when `._loopsFinished` exceeds `.loops`. This is a separate criteria from `not pa_status['Active'] and pa_status['State'] == 0` right?

So we essentially need to decide does `not pa_status['Active'] and pa_status['State'] == 0` supersede the relevance of loops. Or asked in a different way, when the sound file is set to loop, would `not pa_status['Active'] and pa_status['State'] == 0` evaluate to `True` before all loops have finished?

**If the answer is yes:** right now whenever `not pa_status['Active'] and pa_status['State'] == 0` is `True`, we set `self._isFinished` as `True` without accumulating or checking `self._loopsFinished`. So attributes are not updated correctly.

**If the answer is no:** then there is no need to accumulate or keep track of `self._loopsFinished` at all. We just need to check `not pa_status['Active'] and pa_status['State'] == 0` to call `stop()` and set `self._isFinished` as `True`.


Regardless of what you decide how to handle this problem, right now we only enter `_EOS()` when `self._isFinished` is `True`. Then there is no longer the need to call `self._isFinished = True` during `_EOS()` right?

Idk if `_EOS()` itself is a legacy method and can be removed completely. But I have a hunch that without it, looped audio playing won't work properly, and the answer to my question 
> would `not pa_status['Active'] and pa_status['State'] == 0` evaluate to `True` before all loops have finished?
is yes, and we need to handle it carefully in `_checkPlaybackFinished()`.

Reply:
I believe the answer is yes (@mdcutone please correct me if I'm wrong), and I think I follow: If we check finished and it looks like it is finished, we call `_EOS`, which sets `_isFinished` to True if all loops are finished, but if all loops *aren't* finished then `_isFinished` still remains `True`. So the solution should be as simple as adding `else: self._isFinished = False` to `_EOS`? I'll do that now and test it out

Reply:
Perfect! That's it. Somehow I was overcomplicating it in my mind.